homeassistant/components/bmw_connected_drive/sensor.py

```diff
@@ -27,70 +26,84 @@
+    "saved_fuel": ["mdi:fuel", None, VOLUME_LITERS, False],
+    "total_distance": ["mdi:map-marker-distance", None, LENGTH_KILOMETERS, True],
 }
 
 ATTR_TO_HA_IMPERIAL = {
-    "mileage": ["mdi:speedometer", LENGTH_MILES, True],
-    "remaining_range_total": ["mdi:map-marker-distance", LENGTH_MILES, True],
-    "remaining_range_electric": ["mdi:map-marker-distance", LENGTH_MILES, True],
-    "remaining_range_fuel": ["mdi:map-marker-distance", LENGTH_MILES, True],
-    "max_range_electric": ["mdi:map-marker-distance", LENGTH_MILES, True],
-    "remaining_fuel": ["mdi:gas-station", VOLUME_GALLONS, True],
+    # "<ID>": [<MDI_ICON>, <DEVICE_CLASS>, <UNIT_OF_MEASUREMENT>, <ENABLED_BY_DEFAULT>],
+    "mileage": ["mdi:speedometer", None, LENGTH_MILES, True],
+    "remaining_range_total": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
+    "remaining_range_electric": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
+    "remaining_range_fuel": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
+    "max_range_electric": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
+    "remaining_fuel": ["mdi:gas-station", None, VOLUME_GALLONS, True],
     # LastTrip attributes
     "average_combined_consumption": [
         "mdi:flash",
+        None,
         f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
         True,
     ],
     "average_electric_consumption": [
         "mdi:power-plug-outline",
+        None,
         f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
         True,
     ],
     "average_recuperation": [
         "mdi:recycle-variant",
+        None,
         f"{ENERGY_KILO_WATT_HOUR}/100{LENGTH_MILES}",
         True,
     ],
-    "electric_distance": ["mdi:map-marker-distance", LENGTH_MILES, True],
-    "saved_fuel": ["mdi:fuel", VOLUME_GALLONS, False],
-    "total_distance": ["mdi:map-marker-distance", LENGTH_MILES, True],
+    "electric_distance": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
+    "saved_fuel": ["mdi:fuel", None, VOLUME_GALLONS, False],
+    "total_distance": ["mdi:map-marker-distance", None, LENGTH_MILES, True],
 }
 
 ATTR_TO_HA_GENERIC = {
-    "charging_time_remaining": ["mdi:update", TIME_HOURS, True],
-    "charging_status": ["mdi:battery-charging", None, True],
+    # "<ID>": [<MDI_ICON>, <DEVICE_CLASS>, <UNIT_OF_MEASUREMENT>, <ENABLED_BY_DEFAULT>],
+    "charging_time_remaining": ["mdi:update", None, TIME_HOURS, True],
+    "charging_status": ["mdi:battery-charging", None, None, True],
     # No icon as this is dealt with directly as a special case in icon()
-    "charging_level_hv": [None, PERCENTAGE, True],
+    "charging_level_hv": [None, None, PERCENTAGE, True],
     # LastTrip attributes
-    "date_utc": ["mdi:calendar", None, True],
-    "duration": ["mdi:timer-outline", TIME_MINUTES, True],
-    "electric_distance_ratio": ["mdi:percent-outline", PERCENTAGE, False],
+    "date_utc": ["mdi:calendar", DEVICE_CLASS_TIMESTAMP, None, True],
```

Comment:
We don't set a custom icon when setting device class. The frontend should set a custom icon per device class.
```suggestion
    "date_utc": [None, DEVICE_CLASS_TIMESTAMP, None, True],
```